doc/rst/developer/bestPractices/TestSystem.rst

```diff
@@ -370,20 +374,50 @@ instance:
      --x=true # foo.true.good
      --x=false # foo.false.good
 
+
 will compare test output to ``foo.true.good`` for the first execution and
 ``foo.false.good`` for the second.
 
 Any line that is unlabeled will use the default ``.good`` for that test.
 Undefined behavior will occur when both the ``.compopts`` and ``.execopts``
 files specify a ``.good`` file in this way.
 
-Using precomp and prediff files
-+++++++++++++++++++++++++++++++
+If you want to use use default arguments for the test but specify a different
+``.good`` file, you can add a line in your compopts/execopts file as follows
+(note the space before the #):
+
+  .. code-block:: bash
+
+      # foo.execopts
+       # foo.true.good
+
+
+Using precomp, preexec and prediff files
+++++++++++++++++++++++++++++++++++++++++
+
+When creating a ``.precomp``, ``.preexec`` or ``.prediff`` file, the file
+must be an executable. You can turn your script into an executable by running:
+``chmod +x foo.precomp``. For specifying these files for entire directories,
+the files should be named ``PRECOMP``, ``PREXEC``, and ``PREDIFF``,
+respectively.
+
+If you wish to have a system wide ``.prediff`` file, you can use the
+``CHPL_SYSTEM_PREDIFF`` environment variable that takes a comma-separated
+list of prediffs to run before every test.
```

Comment:
prediffs are run on the output after the test is run, maybe "list of prediffs to run after every test before comparing to the good file"